### ISSUE TITLE
Polygon rendering from geojson source - one malformed tile [0.44.1]

### ISSUE THREAD

**Issue by ryanbaumann** [mapbox/mapbox-gl-js#6383]
**mapbox-gl-js version**: 0.44.1

### Steps to Trigger Behavior

 1. Load sample map using geojson source polygon: https://bl.ocks.org/ryanbaumann/fd38662220851abc38681da26437adc6
 2. Zoom in 1 and zoom out one level from the default map position

## Expected behavior

Polygon renders as it does in Geojson.io

## Actual behavior

Polygon has rendering errors at one specific zoom level.

![](https://cl.ly/2N2Z0U1N0D1M/download/Image%202018-03-22%20at%206.54.06%20PM.png)

### Testing

Geojson polygon data - https://gist.github.com/ryanbaumann/fd38662220851abc38681da26437adc6

1. Validated polygon has correct winding order and throws no errors on geojson lint.
![](https://cl.ly/1u420y3C230b/download/Image%202018-03-22%20at%206.58.10%20PM.png)
2. Changing the polygon geometry effects how the issue manifests, but not in a reproducible way.
![screen recording 2018-03-22 at 06 50 pm](https://user-images.githubusercontent.com/4342845/37807355-353660a6-2e02-11e8-8e44-86e4d7c4a625.gif)

**Comment by ryanbaumann** [comment]
Another data point:  Increasing the `buffer` to 200px and reducing the `maxzoom` to 16 setting, I can get the rendering artifact to go away.  

https://bl.ocks.org/ryanbaumann/fd38662220851abc38681da26437adc6/075430fd571757fd3739d99798d3efd9079d94d8

![](https://cl.ly/1G2C403k0s0h/download/Screen%20recording%202018-03-30%20at%2008.29.51%20AM.gif)

**Comment by foundryspatial-duncan** [comment]
I just noticed something similar. I've got a big rectangular polygon with a hole in it to act as a mask layer. At one zoom level (and only one zoom level that I've noticed), this causes what looks like a second polygon to appear.

![screen shot 2018-04-10 at 10 44 11 am](https://user-images.githubusercontent.com/13056053/38582796-ca6e70ec-3cc5-11e8-94ef-13fffa6e1188.png)

The layer is semi-transparent, so the dark purple is where this phantom polygon overlaps with the real one.
